source/agora/node/Ledger.d

```diff
@@ -916,6 +918,44 @@ public class Ledger
         this.fee_man.accumulateFees(block, stakes, &this.utxo_set.peekUTXO);
     }
 
+    /***************************************************************************
+
+        Returns the highest fee among all the transactions which would be
+        considered as a double spent, if `tx` transaction was in the transaction
+        pool.
+
+        If adding `tx` to the transaction pool would not result in double spent
+        transaction, then the return value is Nullable!Amount().
+
+        Params:
+            tx = transaction
+
+        Returns:
+            the highest fee among all the transactions which would be
+            considered as a double spend, if `tx` transaction was in the
+            transaction pool.
+
+    ***************************************************************************/
+
+    public Nullable!Amount getDoubleSpentHighestFee(in Transaction tx) @safe
```

Comment:
Why not return `0` ?

Reply:
it would be a loss of information, having a double spend transaction with 0 fee is not the same as having no duble spend transaction(the case when it is null)